### PR TITLE
filter dataloader calls in tests

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #3221 

# Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests? 
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)?

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/utilities/debugging.py [hunk 1]
```diff
@@ -104,12 +101,18 @@ def track_load_dataloader_call(self, name, dataloaders):
...
+            if strict and dataloader_name == name:
+                return True
+            elif not strict and name in dataloader_name:
+                return True
+            return False
+
+        return [call for call in self.dataloader_sequence_calls if _condition(call['name'])]
+
+    def count_dataloader_calls(self, dataloader_name: str, strict: bool = False) -> int:
+        return len(self.filter_dataloader_calls(dataloader_name, strict))
```
Comment (williamFalcon) [inline pytorch_lightning/utilities/debugging.py:115]: this is exactly the code we want to avoid... it's unclear an unreadable

**Comment by williamFalcon** [comment]
@Borda  i'm sorry... we're done with code that makes things unclear

**Comment by Borda** [comment]
simplified...

**Comment by williamFalcon** [comment]
this is not necessary

**Comment by Borda** [comment]
it is about not duplicate content in several variables - when you have the same content on multiple places it is just open door for bugs...

**Pull request closed without merging**